lib/postgres_v3_experimental.dart

```diff
@@ -0,0 +1,267 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:collection/collection.dart';
+import 'package:stream_channel/stream_channel.dart';
+
+import 'src/v3/connection.dart';
+import 'src/v3/protocol.dart';
+import 'src/v3/query_description.dart';
+import 'src/v3/types.dart';
+
+export 'src/v3/types.dart';
+
+abstract class PgPool implements PgSession, PgSessionExecutor {
+  static Future<PgPool> open(
+    List<PgEndpoint> endpoints, {
+    PgSessionSettings? sessionSettings,
+    PgPoolSettings? poolSettings,
+  }) =>
+      throw UnimplementedError();
+
+  Future<R> withConnection<R>(
+    Future<R> Function(PgConnection connection) fn, {
+    PgSessionSettings? sessionSettings,
+  });
+}
+
+class PgQueryDescription {
+  factory PgQueryDescription.direct(String sql, {List<PgDataType>? types}) =
```

Comment:
I think this will be a frequently used, and should be shorter. Maybe `PgSql(String sql, {...})` for the direct, and `PgSql.map` for the mapped?